code/modules/mob/living/basic/minebots/minebot.dm

```diff
@@ -0,0 +1,179 @@
+	minimum_survivable_temperature = 0
+	health = 125
+	maxHealth = 125
+	melee_damage_lower = 15
+	melee_damage_upper = 15
+	obj_damage = 10
+	attack_verb_continuous = "drills"
+	attack_verb_simple = "drill"
+	attack_sound = 'sound/weapons/circsawhit.ogg'
+	sentience_type = SENTIENCE_MINEBOT
+	speak_emote = list("states")
+	mob_biotypes = MOB_ROBOTIC
+	death_message = "blows apart!"
+	light_system = MOVABLE_LIGHT
+	light_range = 6
+	light_on = FALSE
+	combat_mode = FALSE
+	ai_controller = /datum/ai_controller/basic_controller/minebot
+	///the access card we use to access mining
+	var/obj/item/card/id/access_card
+	///the gun we use to kill
+	var/obj/item/gun/energy/recharge/kinetic_accelerator/minebot/stored_gun
+	///the commands our owner can give us
+	var/list/pet_commands = list(
+		/datum/pet_command/idle,
+		/datum/pet_command/minebot_ability/light,
+		/datum/pet_command/minebot_ability/dump,
+		/datum/pet_command/automate_mining,
+		/datum/pet_command/free/minebot,
+		/datum/pet_command/follow,
+		/datum/pet_command/point_targetting/attack/minebot,
+	)
+
+/mob/living/basic/mining_drone/Initialize(mapload)
+	. = ..()
+
+	var/static/list/death_loot = list(/obj/effect/decal/cleanable/robot_debris)
+	AddElement(/datum/element/death_drops, death_loot)
+	AddElement(/datum/element/footstep, FOOTSTEP_OBJ_ROBOT, 1, -6, sound_vary = TRUE)
+	AddComponent(\
+		/datum/component/tameable,\
+		food_types = list(/obj/item/stack/ore),\
+		tame_chance = 100,\
+		bonus_tame_chance = 5,\
+		after_tame = CALLBACK(src, PROC_REF(activate_bot)),\
+	)
+
+	var/datum/action/cooldown/mob_cooldown/minedrone/toggle_light/toggle_light_action = new(src)
+	var/datum/action/cooldown/mob_cooldown/minedrone/toggle_meson_vision/toggle_meson_vision_action = new(src)
+	var/datum/action/cooldown/mob_cooldown/minedrone/dump_ore/dump_ore_action = new(src)
+	toggle_light_action.Grant(src)
+	toggle_meson_vision_action.Grant(src)
+	dump_ore_action.Grant(src)
+	ai_controller.set_blackboard_key(BB_MINEBOT_LIGHT_ABILITY, toggle_light_action)
+	ai_controller.set_blackboard_key(BB_MINEBOT_DUMP_ABILITY, dump_ore_action)
+
+	stored_gun = new(src)
+	var/obj/item/implant/radio/mining/imp = new(src)
+	imp.implant(src)
+	access_card = new /obj/item/card/id/advanced/gold(src)
+	SSid_access.apply_trim_to_card(access_card, /datum/id_trim/job/shaft_miner)
+
+	RegisterSignal(src, COMSIG_MOB_TRIED_ACCESS, PROC_REF(attempt_access))
+
+/mob/living/basic/mining_drone/set_combat_mode(new_mode, silent = TRUE)
+	. = ..()
+	var/functioning_mode = combat_mode ? "attack" : "collect"
+	icon_state = combat_mode ? "mining_drone_offense" : "mining_drone"
+	to_chat(src, span_info("You are set to [functioning_mode] mode."))
+
+/mob/living/basic/mining_drone/examine(mob/user)
+	. = ..()
+	if(health < maxHealth)
+		if(health >= maxHealth * 0.5)
+			. += span_warning("[p_They()] look slightly dented.")
+		else
+			. += span_boldwarning("[p_They()] look severely dented!")
+
+	if(!stored_gun?.max_mod_capacity)
+		return
```

Comment:
```suggestion
	if(isnull(stored_gun) || !stored_gun.max_mod_capacity)
		return
```

very ambiguous and weird otherwise so let's be safe

Reply:
`isnull()` is more explicit in function (and is also faster in utility), so it should be used for pretty much any typed var in cases like these imo. i do not think that when the code is compiled that it makes any significant difference anyways, it is just far cleaner to have the thing written out rather than do the heaviest of all lifting on the `?` operator